features/facebookImport/src/model/importers/messages-importer.js

```diff
@@ -1,84 +1,22 @@
-    _isJsonMessageFile(entryName) {
+export default class MessagesImporter extends MultipleFilesImporter {
+    _isTargetPostFile(entryName) {
         const formattedEntryName = removeEntryPrefix(entryName);
         return /messages\/(inbox|legacy_threads|message_requests|filtered_threads|archived_threads)\/[0-9_a-z]+\/message_[1-9][0-9]?.json$/.test(
             formattedEntryName
         );
     }
 
-    async _extractJsonEntries(zipFile) {
-        const entries = await relevantZipEntries(zipFile);
-        return entries.filter((fileName) => this._isJsonMessageFile(fileName));
-    }
-
-    async _readJSONFileWithStatus(messageFile, zipFile) {
-        return readFullPathJSONFile(messageFile, zipFile)
-            .then((data) => {
-                return { status: IMPORT_SUCCESS, messageFile, data };
-            })
-            .catch((error) => {
-                return createErrorResult(MessagesImporter, error);
-            });
-    }
-
-    _importMessageThread(facebookAccount, messageThreadResults) {
-        const successfullResults = messageThreadResults.filter(
-            (result) => result.status === IMPORT_SUCCESS
-        );
-
-        for (const each of successfullResults) {
-            const fileName = removeEntryPrefix(each.messageFile);
-            facebookAccount.addImportedFileName(fileName);
-        }
+    _importRawDataResults(facebookAccount, dataResults) {
         facebookAccount.messageThreadsGroup.addMessageThreadsFromData(
-            successfullResults.map((result) => result.data)
+            dataResults
         );
     }
 
-    async _importMessageThreadsFromFiles(
-        messageThreadFiles,
-        zipFile,
-        facebookAccount
-    ) {
-        const messageThreadResults = await Promise.all(
-            messageThreadFiles.map((messageFile) =>
-                this._readJSONFileWithStatus(messageFile, zipFile)
-            )
-        );
-        this._importMessageThread(facebookAccount, messageThreadResults);
-        return messageThreadResults.filter(
-            (result) => !(result.status === IMPORT_SUCCESS)
-        );
-    }
-
-    async import({ zipFile, facebookAccount }) {
-        const messageThreadFiles = await this._extractJsonEntries(zipFile);
-        if (messageThreadFiles.length === 0) {
-            throw new MissingMessagesFilesException();
-        }
-        // TODO: The same message thread can be in multiple files
-        const fileChunks = sliceIntoChunks(messageThreadFiles, 5);
-        const resultChunks = [];
-        for (let currentChunk of fileChunks) {
-            const resultChunk = await this._importMessageThreadsFromFiles(
-                currentChunk,
-                zipFile,
-                facebookAccount
-            );
-
-            resultChunks.push(resultChunk);
-        }
-        const failedResults = resultChunks.flat();
-
-        return failedResults.length > 0 ? failedResults : null;
+    _createMissingFilesError() {
```

Comment:
Do we need this additional indirection for raising an exception?

Reply:
The mechanism that we have for reporting exceptions include now the class name. So we delegate the creation of the exception through the importer to get a concrete exception.